docs/source/architecture.rst

```diff
@@ -0,0 +1,37 @@
+Architecture
+============
+
+StreamAlert consists of multiple AWS components connected together and managed by Terraform.
+
+.. figure:: ../images/sa-complete-arch.png
+  :alt: StreamAlert Architecture
+  :align: center
+  :target: _images/sa-complete-arch.png
+
+Data Lifecycle
+--------------
+
+1. Log data can come through any of the supported `data sources <datasources.html>`_.
+This includes Kinesis, S3, SNS, or using a `StreamAlert App <app-configuration.html>`_ to periodically
```

Comment:
I would use `StreamAlert App Framework`.

Reply:
My understanding is that an `App` is the actual Lambda function which polls data (which I reference here), and the `App Framework` is the set of base classes that you can use to write your own `App`

Reply:
agreed with Austin here, `Framework` is what they're built around, not the thing itself